api/v1alpha1/contour_types.go

```diff
@@ -66,6 +66,13 @@ type ContourSpec struct {
 	//
 	// +kubebuilder:default={name: "projectcontour", removeOnDeletion: false}
 	Namespace NamespaceSpec `json:"namespace,omitempty"`
+
+	// EndpointPublishing is used to publish the network endpoints of Envoy.
+	//
+	// If unset, defaults to LoadBalancerService.
+	//
+	// +kubebuilder:default={type: "LoadBalancerService"}
+	EndpointPublishing EndpointPublishing `json:"endpointPublishing,omitempty"`
```

Comment:
I don't love this name, and I wonder whether we can come up with something more evocative. Data plane strategy? Envoy Service type?

Reply:
@jpeach I agree, I'm not crazy about the name, but I'm struggling with other names. This API is about publishing the network endpoints of Envoy.

Reply:
Updated the struct name to `NetworkPublishing`.